daemon/cluster/listen_addr.go

```diff
@@ -0,0 +1,159 @@
+package cluster
+
+import (
+	"errors"
+	"fmt"
+	"net"
+)
+
+func resolveListenAddr(specifiedInterface, specifiedAddr string) (string, error) {
+	specifiedHost, specifiedPort, err := net.SplitHostPort(specifiedAddr)
+	if err != nil {
+		return specifiedAddr, nil
```

Comment:
I did this on purpose but in retrospect, my reasoning doesn't make sense. I'll change it.